kentikapi/client_integration_test.go

```diff
@@ -417,3 +425,110 @@ func timePtr(v time.Time) *time.Time {
 func durationPtr(v time.Duration) *time.Duration {
 	return &v
 }
+
+func TestGRPCClient_GetAgent(t *testing.T) {
+	tests := []struct {
+		name    string
+		id      string
+		res     *pb.GetAgentResponse
+		errCode codes.Code
+		errMsg  string
+	}{
+		{
+			"invalid request with wrong id",
+			"1000",
+			nil,
+			codes.InvalidArgument,
+			"cannot find ID",
+		},
+		{
+			"valid request with valid id",
+			"968",
+			&pb.GetAgentResponse{Agent: newDummyAgentForGRPC()},
+			codes.OK,
+			"",
+		},
+	}
+
+	ctx := context.Background()
+
+	lis, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+	s := grpc.NewServer()
+	pb.RegisterSyntheticsAdminServiceServer(s, &mockSyntheticsServer{})
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			log.Fatalf("failed to serve: %v", err)
+		}
+	}()
+
+	client := kentikapi.NewClient(kentikapi.Config{
+		SyntheticsGRPCAPIURL: lis.Addr().String(),
+	})
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			request := &pb.GetAgentRequest{Id: tt.id}
+
+			response, err := client.GRPCSyntheticsAdminServiceAPI.GetAgent(ctx, request)
+
+			if response != nil {
+				assert.Equal(t, tt.res.GetAgent(), response.GetAgent())
+			}
+
+			if err != nil {
+				assert.Errorf(t, err, tt.errMsg)
+			}
+		})
+	}
+}
+
+type mockSyntheticsServer struct {
+	pb.UnimplementedSyntheticsAdminServiceServer
+}
+
+func (*mockSyntheticsServer) GetAgent(_ context.Context, req *pb.GetAgentRequest) (*pb.GetAgentResponse, error) {
+	if req.Id == "968" {
```

Comment:
We eventually need ability to set returned response[s] for the mock taking them from test case input. In my opinion this PR is good place for that.

Reply:
Agreed. Let's create whole test infra in the beginning.

Reply:
Done